### PR TITLE
Match pre-commit in dask

### PR DESCRIPTION
**Pull request by jsignell**
This PR copies the pre-commit from dask which seems to work fine

### Motivation
After cleaning out my dev env, I was getting errors from the pre-commit in distributed. I think it has to do with my dev env being python 3.8.

The error looked like:

```
[INFO] Installing environment for https://github.com/ambv/black.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
An unexpected error has occurred: CalledProcessError: command: ('/home/julia/conda/envs/dask-dev/bin/python3.8', '-mvirtualenv', '/home/julia/.cache/pre-commit/repoyh1z3_q8/py_env-python3.7', '-p', 'python3.7')
return code: 1
expected return code: 0
stdout:
    RuntimeError: failed to find interpreter for Builtin discover of python_spec='python3.7'
    
stderr: (none)
Check the log at /home/julia/.cache/pre-commit/pre-commit.log
```

### CODE REVIEW

**Comment by jsignell** [comment]
@jrbourbeau does this seem reasonable?

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,10 +1,11 @@
...
-    rev: stable
-    hooks:
-    - id: black
-      language_version: python3.7
--   repo: https://github.com/pre-commit/pre-commit-hooks
-    rev: v2.3.0
-    hooks:
-    - id: flake8
+  -   repo: https://github.com/python/black
+      rev: stable
```
Comment (jrbourbeau) [inline .pre-commit-config.yaml:3]: I've been running into `black` version mismatches between local and CI dev environments here and in other projects. What do you think about pinning to a specific release of `black`?

```suggestion
      rev: 20.8b1 
```

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -1,10 +1,11 @@
...
+      hooks:
+      - id: black
+        language_version: python3.7
```
Comment (jrbourbeau) [inline .pre-commit-config.yaml:6]:
```suggestion
        language_version: python3
```

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -1,10 +1,11 @@
...
+  -   repo: https://gitlab.com/pycqa/flake8
+      rev: 3.7.9
+      hooks:
+      - id: flake8
+        language_version: python3.7
```
Comment (jrbourbeau) [inline .pre-commit-config.yaml:11]:
```suggestion
        language_version: python3
```

**Review by jrbourbeau — COMMENTED** [review]
Thanks for the PR @jsignell! Apologies for the delayed reply. I'm still running into `RuntimeError: failed to find interpreter for Builtin discover of python_spec='python3.7'` when running locally in a Python 3.8 conda environment. Perhaps we should just [use `python3` for the `language_version`](https://github.com/pre-commit/pre-commit/issues/1375#issuecomment-603906811)? That works for me locally and I suspect is general enough to work for most contributors

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -1,10 +1,11 @@
...
--   repo: https://github.com/ambv/black
-    rev: stable
-    hooks:
-    - id: black
-      language_version: python3.7
--   repo: https://github.com/pre-commit/pre-commit-hooks
-    rev: v2.3.0
-    hooks:
-    - id: flake8
+  -   repo: https://github.com/python/black
```
Comment (jsignell) [inline .pre-commit-config.yaml:2]:
```suggestion
  -   repo: https://github.com/psf/black
```

File: .pre-commit-config.yaml (hunk 1 above)
Comment (jsignell) [reply .pre-commit-config.yaml:3]: I think we should use this pre-commit stuff in ci instead as in https://github.com/dask/dask/pull/6645

File: .pre-commit-config.yaml [hunk 5]
```diff
@@ -1,10 +1,11 @@
...
+      rev: stable
+      hooks:
+      - id: black
+        language_version: python3.7
+  -   repo: https://gitlab.com/pycqa/flake8
+      rev: 3.7.9
```
Comment (jsignell) [inline .pre-commit-config.yaml:8]:
```suggestion
      rev: 3.8.3
```

File: .travis.yml [hunk 6]
```diff
@@ -16,7 +16,6 @@ matrix:
   - os: linux
     language: python
     python: 3.6
```
Comment (jrbourbeau) [inline .travis.yml:18]: I _think_ we can remove this job altogether as previously it was just running `flake8` and `black` (for an example build see https://travis-ci.org/github/dask/distributed/jobs/727483894)

**Review by jrbourbeau — COMMENTED** [review]
Thanks @jsignell! Overall this looks really good to me, just left one small comment about the changes to `.travis.yml`

**Comment by jsignell** [comment]
Ok I merged a non-blacked change to make sure this fails properly.

**Comment by jsignell** [comment]
@jrbourbeau I did `pre-commit run --all-files` locally as recommended by the test failure:https://github.com/dask/distributed/runs/1124209877 and I got a big diff. Would you prefer that in a follow on PR?

**Comment by jsignell** [comment]
I actually can't quite get the same thing to happen on master so maybe things are ok there?

**Comment by jsignell** [comment]
Yeah I think https://github.com/dask/distributed/pull/4081 fixed the formatting on master so this PR should be good to go @jrbourbeau.

**Review by jrbourbeau — APPROVED** [review]
Woo, thanks @jsignell!

**Pull request merged**